tests/test_virtuaaliviivakoodi_deconstruct.py

```diff
@@ -0,0 +1,284 @@
+                        "reference": "RF6078777679656628687",
+                        "euro_amount": decimal.Decimal("935.85"),
+                        "due_date": None,
+                        "version": SymbolVersion.VERSION_5.value
+                    },
+                    "516800014000502670009358560000078777679656628687000000"
+            ),
+            (
+                    {
+                        "iban": "FI7331313001000058",
+                        "reference": "RF10868624",
+                        "euro_amount": decimal.Decimal("0.0"),
+                        "due_date":  date(2013, 8, 9),
+                        "version": SymbolVersion.VERSION_5.value
+                    },
+                    "573313130010000580000000010000000000000000868624130809"
+            ),
+            (
+                    {
+                        "iban": "FI8333010001100775",
+                        "reference": "RF7192125374252539897737",
+                        "euro_amount": decimal.Decimal("150000.20"),
+                        "due_date": date(2016, 5, 25),
+                        "version": SymbolVersion.VERSION_5.value
+                    },
+                    "583330100011007751500002071092125374252539897737160525"
+            ),
+            (
+                    {
+                        "iban": "FI3936363002092492",
+                        "reference": "RF66590738390",
+                        "euro_amount": decimal.Decimal("1.03"),
+                        "due_date": date(2023, 3, 11),
+                        "version": SymbolVersion.VERSION_5.value
+                    },
+                    "539363630020924920000010366000000000000590738390230311"
+            ),
+            (
+                    {
+                        "iban": "FI9239390001003391",
+                        "reference": "RF951357914",
+                        "euro_amount": decimal.Decimal("0.02"),
+                        "due_date": date(2099, 12, 24),
+                        "version": SymbolVersion.VERSION_5.value
+                    },
+                    "592393900010033910000000295000000000000001357914991224"
+            ),
+        ]
+    )
+    def test_valid_virtuaaliviivakoodis_deconstruct(self, expected_result, input):
+
+        version, iban, euro_amount, reference, due_date = deconstruct_virtuaaliviivakoodi(input)
+
+        result = {
+            "version": version,
+            "iban": iban,
+            "euro_amount": euro_amount,
+            "reference": reference,
+            "due_date": due_date,
+        }
+
+        self.assertEqual(result, expected_result)
+
+    @parameterized.expand([
+        (
+                "692393900010033910000000295000000000000001357914991224",
+                InvalidVersionException
+         ),
+        (
+                "392393900010033910000000295000000000000001357914991224",
+                InvalidVersionException
+        )
+    ])
```

Comment:
Some test cases to throw in:
- Too long or too short virtuaaliviivakoodi
- Faulty due date e.g. 999999

Reply:
Added the test cases for too short and long virtuaaliviivakoodi and a new exception for it. Also added the invalid date but didn't add a new exception for it since the exception originates from the datetime module.